framework/src/media/InputHandler.cpp

```diff
@@ -71,12 +72,118 @@ bool InputHandler::doStandBy()
 	return true;
 }
 
+bool InputHandler::open()
+{
+	if (!mInputDataSource) {
+		meddbg("mInputDataSource is nullptr!\n");
+		return false;
+	}
+
+	if (!getStreamBuffer()) {
+		auto streamBuffer = StreamBuffer::Builder()
+							.setBufferSize(CONFIG_HANDLER_STREAM_BUFFER_SIZE)
+							.setThreshold(CONFIG_HANDLER_STREAM_BUFFER_THRESHOLD)
+							.build();
+		if (!streamBuffer) {
+			meddbg("streamBuffer is nullptr!\n");
+			return false;
+		}
+		setStreamBuffer(streamBuffer);
+	}
+
+	// Open data soruce
+	if (!(mInputDataSource->isPrepared() || mInputDataSource->open())) {
```

Comment:
`please test focus manager, if focus gain after lost, app should prepare media player again.`
Yes, app called media player prepare(Async) again, but because player's state is not PLAYER_STATE_CONFIGURED, prepare(Async) again failed (in other words, it's not allowed to do prepare at PLAYER_STATE_PAUSED state). should we change the logic?
Anyway, InputHandler::open's implementaion supports being called again to wait data buffering (reach threshold).